lib/core/OutgoingWebSocket.js

```diff
@@ -107,6 +109,20 @@ class OutgoingWebSocket extends EventEmitter {
       this.emit(kEnableIncomingQueue);
       return;
     }
+
+    if(customRequestEnabled) {
+      const customReqInstance = CustomRequestHandler.getInstance();
+      let resp;
+      try {
+        resp = JSON.parse(msg);
+        if (resp && customReqInstance.getList().hasOwnProperty(resp.id)) {
+          customReqInstance.customRequestList[resp.id].resolve(msg)
+          return;
+        }
+      } catch (error) {
+        console.error('Error parsing JSON:', error);
```

Comment:
check for message if message includes id or not if message includes the id we have sent don't emit the message. This is just an extra validation if its possible to add. We don't want to emit the message to client if the command was triggered by us in case parsing breaks ideally we know it should not fail but to ensure have that check if msg has id.

Reply:
This we can't do without parsing as it always returns a json string. Unless we do some regex parsing which will be slow.